core/src/main/java/com/datastax/oss/driver/internal/core/cql/CqlPrepareAsyncProcessor.java

```diff
@@ -45,12 +51,22 @@ public boolean canProcess(Request request, GenericType<?> resultType) {
   }
 
   @Override
-  public RequestHandler<PrepareRequest, CompletionStage<PreparedStatement>> newHandler(
+  public CompletionStage<PreparedStatement> process(
       PrepareRequest request,
       DefaultSession session,
       InternalDriverContext context,
       String sessionLogPrefix) {
-    return new CqlPrepareAsyncHandler(
-        request, preparedStatementsCache, session, context, sessionLogPrefix);
+
+    try {
+      return cache.get(
+          request,
+          () -> new CqlPrepareHandler(request, session, context, sessionLogPrefix).handle());
```

Comment:
it looks like failed results are cached, and all subsequent prepares for that query would fail.  Test showing that here: https://gist.github.com/tolbertam/4f0581f2fbb3b90f5b79968e3a6f173d